src/ZODB/Connection.py

```diff
@@ -884,7 +884,10 @@ def open(self, transaction_manager=None, delegate=True):
         """
 
         if transaction_manager is None:
-            transaction_manager = transaction.manager
+            # The .manager bit below unwraps the threaded
+            # manager so we can call unregisterSynch in close
+            # when close is called from another thread.
+            transaction_manager = transaction.manager.manager
```

Comment:
This sort of peeking into the implementation details of another package demands a versioned dependency requirement in `setup.py` IMHO.